include/Gaffer/TweakPlug.h

```diff
@@ -123,30 +123,51 @@ class GAFFER_API TweakPlug : public Gaffer::ValuePlug
 			MissingMode missingMode = MissingMode::Error
 		) const;
 
+		template< typename T >
+		static T applyValueTweak(
+			const T &source,
+			const T &tweak,
+			TweakPlug::Mode mode,
+			const std::string &tweakName
+		);
```

Comment:
I'm not super comfortable with this way of factoring things, for a few reasons :

- It leaves us duplicating a lot of the TweakPlug logic in a separate component, and that logic is not the most straightforward.
- I think other TweakPlug clients could plausibly benefit from array operations (masked or unmasked).
- We've managed to serve the needs of lots of different clients with a single API so far, and that API has successfully encapsulated all of the complexity (although I admit ShaderTweaks isn't as straightforward).

So, questions :

- Could the existing `applyTweak()` function be modified to do non-masked operations on arrays, benefiting all existing clients?
- Could an additional masked operation abstract away all the stuff to do with primitive variables? Maybe something like the below?

```
/// As above, but applying the tweak to individual elements of VectorTypedData,
/// as specified by `mask`.
template<class GetDataFunctor, class SetDataFunctor>
bool applyMaskedTweak(
	GetDataFunctor &&getDataFunctor,
	SetDataFunctor &&setDataFunctor,
        const vector<bool> &mask,
        // Size of array to make for `Create` mode.
        size_t createSize,
	MissingMode missingMode = MissingMode::Error
) const;
```